### PR TITLE
test(profiling): rework gevent task ignore test

### PR DESCRIPTION
**Pull request by jd**
The existing test works fine in 99% of the case, but since the fib() call is a
bit too short and the while loop timing a bit too fast, in some circumstances,
it might take several minutes for the profiler to catch it.

This extends the time needed to run fib() to have a better change to catch the
task, and refactor the opposite test so the code is shared across the both
positive and negative tests.

I ran this test in a loop a few hundreds of time and it never failed. 🤞

### CODE REVIEW

File: tests/profiling/collector/test_stack.py [hunk 1]
```diff
@@ -154,44 +156,28 @@ def test_no_ignore_profiler_single():
...
     # sure to catch it with the StackProfiler and that it's not ignored.
     c = CollectorTest(p._profiler._recorder, interval=0.00001)
     c.start()
+    cond = operator.not_ if ignore else lambda x: x
     # Wait forever and stop when we finally find an event with our collector task id
     while True:
         events = p._profiler._recorder.reset()
-        if c._worker.ident in {e.task_id for e in events[stack.StackSampleEvent]}:
+        ids = {e.task_id for e in events[stack.StackSampleEvent]}
+        if cond(c._worker.ident in ids):
```
Comment (P403n1x87) [inline tests/profiling/collector/test_stack.py:180]:
```suggestion
        if ignore != (c._worker.ident in ids):
```

**Review by brettlangdon — APPROVED** [review]

File: ddtrace/internal/periodic.py [hunk 2]
```diff
@@ -228,6 +228,6 @@ def join(
     def on_shutdown():
         pass
 
-    @staticmethod
-    def periodic():
+    def periodic(self):
```
Comment (Kyle-Verhoog) [inline ddtrace/internal/periodic.py:231]: unrelated?

File: ddtrace/profiling/collector/__init__.py [hunk 3]
```diff
@@ -33,15 +37,16 @@ class PeriodicCollector(Collector, periodic.PeriodicService):
...
 
     def periodic(self):
+        # type: (...) -> None
         """Collect events and push them into the recorder."""
         for events in self.collect():
             self.recorder.push_events(events)
 
-    @staticmethod
-    def collect():
+    def collect(self):
```
Comment (Kyle-Verhoog) [inline ddtrace/profiling/collector/__init__.py:45]: same as above?

File: ddtrace/internal/periodic.py (hunk 2 above)
Comment (jd) [reply ddtrace/internal/periodic.py:231]: well it is otherwise mypy complains :)

File: ddtrace/profiling/collector/__init__.py (hunk 3 above)
Comment (jd) [reply ddtrace/profiling/collector/__init__.py:45]: same as above!

File: ddtrace/internal/periodic.py (hunk 2 above)
Comment (Kyle-Verhoog) [reply ddtrace/internal/periodic.py:231]: Ah missed that a typehint was added in the test case :+1:

**Review by Kyle-Verhoog — APPROVED** [review]
